ackhandler/received_packet_handler.go

```diff
@@ -39,31 +39,31 @@ func (h *receivedPacketHandler) ReceivedPacket(packetNumber protocol.PacketNumbe
 		return errInvalidPacketNumber
 	}
 
-	if packetNumber > h.ignorePacketsBelow {
-		if err := h.packetHistory.ReceivedPacket(packetNumber); err != nil {
-			return err
-		}
-	}
-
 	if packetNumber > h.largestObserved {
 		h.largestObserved = packetNumber
 		h.largestObservedReceivedTime = time.Now()
 	}
 
+	if packetNumber <= h.lowerLimit {
+		return nil
+	}
+
+	if err := h.packetHistory.ReceivedPacket(packetNumber); err != nil {
+		return err
+	}
 	h.maybeQueueAck(packetNumber, shouldInstigateAck)
 	return nil
 }
 
-func (h *receivedPacketHandler) ReceivedStopWaiting(f *frames.StopWaitingFrame) error {
-	// ignore if StopWaiting is unneeded, because we already received a StopWaiting with a higher LeastUnacked
-	if h.ignorePacketsBelow >= f.LeastUnacked {
-		return nil
+// SetLowerLimit sets a lower limit for acking packets
+// packets with packet numbers smaller than p will not be acked
+func (h *receivedPacketHandler) SetLowerLimit(p protocol.PacketNumber) {
+	// TODO: this check is not needed anymore once we drop support for STOP_WAITINGs
+	if p == 0 {
+		return
 	}
-
-	h.ignorePacketsBelow = f.LeastUnacked - 1
-
-	h.packetHistory.DeleteBelow(f.LeastUnacked)
-	return nil
+	h.lowerLimit = p - 1
```

Comment:
This looks weird – if we have a function SetLowerLimit and a var lowerLimit I'd expect them to deal with the same value. Can we remove this here and change the comparison above?